### ISSUE TITLE
Workflows not in repository root are not detected

### ISSUE THREAD

**Issue by AndreyNautilus** [rhysd/actionlint#242]
`actionlint` can't find workflows in the non-root folder of the repository if there is no `.git` file or folder next to `.github` folder.

**How to reproduce:**
I have a repository with the following structure:
```
/
├─ .github/
│  ├─ workflows/
│  │  ├─ *.yaml workflows for this repository
├─ template1/
│  ├─ .github/
│  │  ├─ workflows/
│  │  │  ├─ *.yaml workflows for template1
├─ template2/
   ├─ github/
      ├─ workflows/
         ├─ *.yaml workflows for template2
```
I use this repository to automatically generate workflows for newly created repositories in my organization (`template1` can be "python project" and `template2` can be "C++ project").

I can easily lint workflows in `.github/workflows` in the root of the repository (using GitHub Acitons):
```yaml
- run: actionlint
```
I also want to lint workflows in the templates folder, like:
```yaml
- run: actionlint
  working-directory: template1
- run: actionlint
  working-directory: template2
```
but this doesn't work. `actionlint` simply doesn't detect `template1` and `template2` as "projects", because of this check: https://github.com/rhysd/actionlint/blob/acc6e8cd8e2c25c2220da68465a968c499bd1207/project.go#L29-L32 
This seems like a bug.

**Workarounds:**
1. automatically create (and then remove) an empty `.git` file next to `.github` folder;
2. use `find` to find workflows I need, `cat` them and pass to `actionlint`.

I can workaround the issue, but it would be great if it is fixed in the code.

**Comment by rhysd** [comment]
How can actionlint determine the YAML files put in the `template1` and `template2` are workflow files for GitHub Actions? `.github/workflows` is the only official space to put GitHub Actions workflow files.

> 1. automatically create (and then remove) an empty .git file next to .github folder;
> 2. use find to find workflows I need, cat them and pass to actionlint.

Passing the workflow file paths as arguments is better.

```
actionlint ./template1/github/workflows/*.yaml ./template2/github/workflows/*.yaml
```

**Comment by AndreyNautilus** [comment]
> How can actionlint determine the YAML files put in the `template1` and `template2` are workflow files for GitHub Actions? `.github/workflows` is the only official place to put GitHub Actions workflow files.

Based on the path. If `.github/workflows` dir exists in the current folder, then treat all `.github/workflows/*.yaml` files as workflows (simply drop the check for `.git` file/folder).
My use case is slightly more complicated: `template1` and `template2` are [cookiecutter](https://github.com/cookiecutter/cookiecutter) templates, they contain jinja templates for workflows, not the workflows themselves. I call cookiecutter on CI and instantiate the templates (to verify that instantiation works). I wanted to lint the generated files to verify that they are valid workflows, but actionlint didn't find errors even though the errors were there. And it took me quite some time (including reading the sources) to figure out why actionlint doesn't lint the generated files and instead it lints workflows in the repository root `.github` folder. It's just non-intuitive.

The [documentation](https://rhysd.github.io/actionlint/usage.html) says:
> To check all workflow files in the current repository, just run **actionlint** without arguments. It automatically finds the _nearest_ **.github/workflows** directory:

It says "nearest .github/workflows directory" it doesn't say that the directory must be in the repository root - that's why I thought that my use case should work out of the box.
`actionlint` makes an impression of a linter that works with _files_, not with _repositories_.

Another inconvenience of the current implementation: if I create a github action that runs actionlint _for a specific directory_ (a dedicated action is useful to store "generic" `actionlint.yaml` configuration with all custom self-hosted labels) then I can't let actionlint to "autodetect" workflows anymore, I need to reimplement this autodetection myself.

> Passing the workflow file paths as arguments is better.

Thanks for the tip, yes this is much better.

To summarise: this is a minor inconvenience which can be worked around, but documentation should explain this corner case.

BTW, thanks for the linter! It already saved quite some time in my work!

**Comment by Morgahl** [comment]
Globbing in the manner described does not appear to work in the Docker base GH Actions workflow described here: https://github.com/rhysd/actionlint/blob/main/docs/usage.md#using-actionlint-from-go-program

This appears to expect the shell to expand the glob in the current usage. Would it be viable to add glob expansion to the binary itself via: https://pkg.go.dev/path/filepath#Glob.
